lib/active_resource/base.rb

```diff
@@ -652,11 +652,11 @@ def connection(refresh = false)
       end
 
       def headers
-        self._headers ||= {}
+        headers_state = self._headers || {}
         if superclass != Object && superclass.headers
```

Comment:
would not `superclass.headers` always return true? I don't see when this condition is not true.